client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java

```diff
@@ -126,11 +124,12 @@ public void onFailure(Exception e) {
             }
         };
         highLevelClient().getAllPitsAsync(RequestOptions.DEFAULT, getPitsListener);
-        highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
-        assertTrue(countDownLatch.await(10, TimeUnit.SECONDS));
+
         // validate no pits case
-        getAllPitResponse = highLevelClient().getAllPits(RequestOptions.DEFAULT);
-        assertTrue(getAllPitResponse.getPitInfos().size() == 0);
-        highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
+        assertBusy(() -> {
+            GetAllPitNodesResponse getAllPitResponse1 = highLevelClient().getAllPits(RequestOptions.DEFAULT);
+            assertTrue(getAllPitResponse1.getPitInfos().size() == 0);
+            highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
```

Comment:
What was wrong with the previous single collection + delete, doesn't `deleteAllPitsAsync` have a deterministic conclusion, shouldn't all the Pit be deleted after only one call?

This  smells like there might be an implementation bug, can you help me understand?

Reply:
no there is a race condition between get and delete, so if delete finishes first , get pits assert fails leading to flaky tests

Reply:
Also another fix is i removed countdown latch with assertbusy , both are technically the same as wait time is at max 10 seconds - this was not needed but did it to make the code look clean.